### PR TITLE
feat: add AvatarGroup component

### PR DESCRIPTION
**Pull request by langz**
This PR adds the AvatarGroup component, which tries to nicely group and style the grouping of the avatars.
Take a look here https://eufemia-featavatargroup.gtsb.io/uilib/components/avatar#grouping-avatars

### CODE REVIEW

File: packages/dnb-eufemia/src/components/avatar/style/_avatar.scss [hunk 1]
```diff
@@ -87,4 +87,41 @@
       line-height: var(--avatar-line-height--x-large);
     }
   }
+
+  &--group {
+    display: flex;
+    justify-content: left;
```
Comment (langz) [inline packages/dnb-eufemia/src/components/avatar/style/_avatar.scss:93]: Don't think I need this 🤔

File: packages/dnb-eufemia/src/components/avatar/AvatarGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    className,
+    children: childrenProp,
+    variant,
+    size,
+    maxElements: maxElementsProp,
+    ...props
+  } = extendPropsWithContext(
+    { ...defaultProps, ...localProps },
+    defaultProps,
+    context?.Avatar
```
Comment (thaytharma) [inline packages/dnb-eufemia/src/components/avatar/AvatarGroup.tsx:67]:
```suggestion
    context?.AvatarGroup
```

Not 100% sure here.

File: packages/dnb-eufemia/src/components/avatar/style/_avatar.scss [hunk 3]
```diff
@@ -87,4 +87,41 @@
...
   }
+
+  &--group {
+    display: flex;
+    justify-content: flex-end;
+    flex-direction: row-reverse;
+    .dnb-avatar {
+      box-sizing: content-box;
+
+      // size small
```
Comment (thaytharma) [inline packages/dnb-eufemia/src/components/avatar/style/_avatar.scss:98]: Personally, I don't think this comment adds any value as the selector name makes it self-explanatory.
Same for the other sizes. What do you think? 🤔

Comment (langz) [reply packages/dnb-eufemia/src/components/avatar/style/_avatar.scss:98]: I agree. 
I just tried to follow an existing pattern that I've seen around in other components :)

File: packages/dnb-eufemia/src/components/avatar/AvatarGroup.tsx (hunk 2 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/avatar/AvatarGroup.tsx:67]: Thanks, I'll fix this :)

File: packages/dnb-eufemia/src/components/avatar/style/_avatar.scss (hunk 3 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/avatar/style/_avatar.scss:98]: I'll remove these comments :)

File: packages/dnb-eufemia/src/components/avatar/AvatarGroup.tsx (hunk 2 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/avatar/AvatarGroup.tsx]: Done

**Review by langz — COMMENTED** [review]
Looks okay to me 🙏

File: packages/dnb-design-system-portal/src/docs/uilib/components/avatar/properties.md [hunk 4]
```diff
@@ -17,3 +17,14 @@ showTabs: true
...
 | [Space](/uilib/components/space/properties) | _(optional)_ spacing properties like `top` or `bottom` are supported.                                                             |
+
+### `Avatar.Group` properties
+
+| Properties                                  | Description                                                                                                                                    |
+| ------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| `size`                                      | _(optional)_ Size of the Avatars, and "elements hidden text (+x)". Options: `small` \| `medium` \| `large` \| `x-large`. Defaults to `medium`. |
+| `variant`                                   | _(optional)_ Override the variant of the Avatars. Options: `primary` \| `secondary` \| `tertiary`. Defaults to `primary`.                      |
+| `maxElements`                               | _(optional)_ Number of max displayed elements, including the "elements hidden text (+x)".                                                      |
```
Comment (thaytharma) [inline packages/dnb-design-system-portal/src/docs/uilib/components/avatar/properties.md:27]:
```suggestion
| `maxElements`                               | _(optional)_ Number of max displayed elements, including the "elements hidden text (+x)". Defaults to 4.                                                     |
```

**Comment by thaytharma** [comment]
> Looks okay to me 🙏

I think it looks great! 🚀

**Review by thaytharma — APPROVED** [review]
I think it looks great! 🚀

**Pull request merged**